torch_xla_py/xla_dist.py

```diff
@@ -6,24 +6,42 @@
 
 class Worker(object):
 
-  def __init__(self, internal_ip, mach_type, zone, hostname=None):
+  def __init__(self, internal_ip, mach_type, zone):
     self._internal_ip = internal_ip
     self._mach_type = mach_type
     self._zone = zone
+
+
+class ClientWorker(Worker):
+
+  def __init__(self, internal_ip, machine_type, zone, hostname=None):
+    super(ClientWorker, self).__init__(internal_ip, machine_type, zone)
     self._hostname = hostname
 
 
+class ServiceWorker(Worker):
+  # Same as base Worker ATM.
+  pass
+
+
 class Cluster(object):
 
-  def __init__(self, workers):
-    self._workers = workers
+  def __init__(self, client_workers, service_workers):
+    """Creates a cluster object.
+
+    Args:
+      client_workers: a list of ClientWorker objects.
+      service_workers: a list of ServiceWorker objects.
+    """
+    self._client_workers = client_workers
```

Comment:
I'd do a `list(...)` here, to make sure that if the caller pass us a list object he later modifies, it does not affect our copy.

Reply:
Good idea. Done.